### PR TITLE
Allow updating existing repo:branch

### PR DESCRIPTION
**Pull request by jpopelka**
Also, discover branch to onboard/update if not specified.

### CODE REVIEW

File: onboard/onboard.py [hunk 1]
```diff
@@ -63,10 +62,34 @@ def create_sg_repo(self, pkg_name: str) -> GitProject:
...
+    @staticmethod
+    def get_distgit_branch(pkg_name: str) -> Optional[str]:
+        dg_token = getenv("DISTGIT_TOKEN")
+        if not dg_token:
+            logger.error("No DISTGIT_TOKEN specified.")
+            return None
+        service = PagureService(token=dg_token, instance_url="https://git.centos.org/")
+        dg_project = service.get_project(namespace="rpms", repo=pkg_name)
+        branches = dg_project.get_branches()
+        for b in ["c8s", "c8"]:
```
Comment (TomasTomecek) [inline onboard/onboard.py]: please add c8s-stream-rhel

File: onboard/onboard.py [hunk 2]
```diff
@@ -63,10 +62,34 @@ def create_sg_repo(self, pkg_name: str) -> GitProject:
...
+                return b
+        else:
+            logger.error("No c8s or c8 branches in dist-git.")
+            return None
+
+    def run(self, pkg_name: str, branch: str = None, skip_build: bool = False):
         action = "Updating" if self.update else "Onboarding"
+
+        if not branch:
+            logger.info("No branch specified. Discovering dist-git.")
```
Comment (TomasTomecek) [inline onboard/onboard.py]: nitpick: I'd say "Inspecting" should be a better word - I understand you "discover" things which are not known to people, we know what dist-git is.

File: onboard/onboard.py (hunk 1 above)
Comment (jpopelka) [reply onboard/onboard.py]: added

File: onboard/onboard.py (hunk 2 above)
Comment (jpopelka) [reply onboard/onboard.py]: changed

**Review by TomasTomecek — APPROVED** [review]
LGTM

**Pull request merged**